ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -1142,4 +1150,240 @@ public void shouldThrowOnNonIntegerId() {
+    final String query =
+        "CREATE STREAM X AS\n" +
+        "  SELECT * FROM TEST1;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateStreamAsSelect result = (CreateStreamAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(2)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of("  SELECT * FROM TEST1".length())));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCt() {
+    // Given:
+    final String query = "CREATE TABLE X WITH (kafka_topic='X');";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateTable result = (CreateTable) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(1)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of(query.length() - 1)));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCtas() {
+    // Given:
+    final String query = "CREATE TABLE X AS SELECT COUNT(1) FROM TEST1 GROUP BY ROWKEY;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateTableAsSelect result = (CreateTableAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(1)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of(query.length() - 1)));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsSimpleCtasMultiLine() {
+    // Given:
+    final String query =
+        "CREATE TABLE X AS\n" +
+        "  SELECT COUNT(1)\n" +
+        "  FROM TEST1\n" +
+        "  GROUP BY ROWKEY;";
+    final SingleStatementContext stmt = givenQuery(query);
+
+    // When:
+    final CreateTableAsSelect result = (CreateTableAsSelect) builder.buildStatement(stmt);
+
+    // Then:
+    final NodeLocation loc = result.getLocation().get();
+    assertThat(loc.getStartLineNumber(), is(1));
+    assertThat(loc.getStartColumnNumber(), is(1));
+    assertThat(loc.getStopLine(), is(OptionalInt.of(4)));
+    assertThat(loc.getStopColumnNumber(), is(OptionalInt.of("  GROUP BY ROWKEY".length())));
+    assertThat(loc.getLength(), is(OptionalInt.of(query.length() - 1)));
+  }
+
+  @Test
+  public void shouldGetCorrectLocationsComplexCtas() {
```

Comment:
I think this test is really helpful to prove to yourself that the code works, but in similar vein to above I don't think it's a very good unit test to check in.

Instead, what we're looking for here, is to see that we're correctly setting these values for internal things (the query component, a join component, etc...) do we need to test all of them? if so, they should each be separate tests - but I don't think we need to test all of them. Instead let's just check for one or two "interesting" sub-components. That way, you or I (as the reviewer) can review and make sure that the numbers you're passing into the asserts actually make sense